go/vt/vtgate/engine/transaction_status.go

```diff
@@ -76,25 +77,37 @@ func (t *TransactionStatus) getFields() []*querypb.Field {
 }
 
 func (t *TransactionStatus) TryExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool) (*sqltypes.Result, error) {
-	transactionState, err := vcursor.ReadTransaction(ctx, t.TransactionID)
+	var transactionStatuses []*querypb.TransactionMetadata
+	var err error
+	if t.TransactionID != "" {
+		var transactionState *querypb.TransactionMetadata
+		transactionState, err = vcursor.ReadTransaction(ctx, t.TransactionID)
+		transactionStatuses = append(transactionStatuses, transactionState)
```

Comment:
mostly not an issue, but there can be an error as well, so we will append a nil to the slice.

Reply:
yeah, but before we use the slice, we check for the error and return.